### PR TITLE
Refactor FXIOS-7684 [v121] Update Amazon engines and remove out of contract ones

### PR DESCRIPTION
**Pull request by mkaply**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-7684)
[GitHub issue](https://github.com/mozilla-mobile/firefox-ios/issues/17136)

## :bulb: Description
Changing Amazon URLS and removing out of contract engines

## :pencil: Checklist
You have to check all boxes before merging
- [X] Filled in the above information (tickets numbers and description of your work)
- [X] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [X] Wrote unit tests and/or ensured the tests suite is passing
- [X] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [X] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

**Review by cyndichin — COMMENTED** [review]
looks good to me, but I will add @nbhasin2 as the reviewer as he has more context, thanks!

**Review by nbhasin2 — APPROVED** [review]

File: Client/Assets/Search/SearchPlugins/amazon-au.xml [hunk 1]
```diff
@@ -5,11 +5,8 @@
...
-<Url type="text/html" method="GET" template="https://www.amazon.com.au/exec/obidos/external-search/">
-  <Param name="field-keywords" value="{searchTerms}"/>
-  <Param name="mode" value="blended"/>
-  <Param name="tag" value="moz-au-mbl-22"/>
-  <Param name="sourceid" value="Mozilla-search"/>
+<Url type="text/html" method="GET" template="https://www.amazon.com.au/s">
+  <Param name="k" value="{searchTerms}"/>
```
Comment (nbhasin2) [inline Client/Assets/Search/SearchPlugins/amazon-au.xml:9]: nit: Are we okay removing all the other sourceid for Mozilla search here?

Comment (nbhasin2) [reply Client/Assets/Search/SearchPlugins/amazon-au.xml:9]: @mkaply this is a generic comment considering we did for the one's below so I am assuming its safe to do so

**Comment by nbhasin2** [comment]
@Mergifyio backport release/v121

**Pull request merged**